### PR TITLE
[GOBBLIN-719] fix invalid git links for classes in docs

### PR DESCRIPTION
**Pull request by jhsenjaliya**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-XXX

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
gobblin docs had some invalid links pointing to LinkedIn repo, this PR not only fixes the repo location but also fixes the file/class location in cases where it has changed.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
no test case required since the changes are in docs

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by yukuai518** [comment]
+1 
Thanks for fixing all the link issues!

**Comment by jhsenjaliya** [comment]
@yukuai518 , can you merge the changes, I need to update docs as part of #2578 and it will hekop to avoid conflicts. Thanks

**Comment by yukuai518** [comment]
I dont have previllege to merge. @Hung Tran<mailto:hutran@linkedin.com>  please help Jay with his pr.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

File: gobblin-docs/case-studies/Kafka-HDFS-Ingestion.md [hunk 1]
```diff
@@ -107,17 +107,17 @@ After the job finishes, the job output file will be in `/gobblintest/job-output/
...
 
 ## Writer and Publisher
 
-Any desired writer and publisher can be used, e.g., one may use the [`AvroHdfsDataWriter`](https://github.com/linkedin/gobblin/blob/master/gobblin-core/src/main/java/gobblin/writer/AvroHdfsDataWriter.java) and the [`BaseDataPublisher`](https://github.com/linkedin/gobblin/blob/master/gobblin-core/src/main/java/gobblin/publisher/BaseDataPublisher.java), similar as the [Wikipedia example job](../Getting-Started). If plain text output file is desired, one may use [`SimpleDataWriter`](https://github.com/linkedin/gobblin/blob/master/gobblin-core/src/main/java/gobblin/writer/SimpleDataWriter.java).
```
Comment (csoseman) [inline gobblin-docs/case-studies/Kafka-HDFS-Ingestion.md]: We should probably update the `Wikipedia example job` link to https://github.com/apache/incubator-gobblin/blob/master/gobblin-example/src/main/resources/wikipedia.pull. The current link makes you think we should go to `readthedocs` but I don't think that Getting-Started page would be extremely helpful in this case.

File: gobblin-docs/metrics/Existing-Reporters.md [hunk 2]
```diff
@@ -6,15 +6,15 @@ Table of Contents
...
+* [Output Stream Event Reporter](https://github.com/apache/incubator-gobblin/blob/master/gobblin-metrics-libs/gobblin-metrics-base/src/main/java/org/apache/gobblin/metrics/reporter/OutputStreamEventReporter.java): Emits events to any output stream, including STDOUT and files.
+* [Kafka Event Reporter](https://github.com/apache/incubator-gobblin/blob/master/gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/metrics/kafka/KafkaEventReporter.java): Emits events to Kafka topic as Json messages.
+* [Kafka Avro Event Reporter](https://github.com/apache/incubator-gobblin/blob/master/gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/metrics/kafka/KafkaAvroEventReporter.java): Emits events to Kafka topic as Avro messages using the schema [GobblinTrackingEvent](https://github.com/linkedin/gobblin/blob/master/gobblin-metrics/src/main/avro/GobblinTrackingEvent.avsc).
```
Comment (csoseman) [inline gobblin-docs/metrics/Existing-Reporters.md]: `GobblinTrackingEvent` should be changed to https://github.com/apache/incubator-gobblin/blob/master/gobblin-metrics-libs/gobblin-metrics-base/src/main/avro/GobblinTrackingEvent.avsc

File: gobblin-docs/metrics/Gobblin-Metrics-Architecture.md [hunk 3]
```diff
@@ -19,15 +19,15 @@ Each metric context contains the following instance variables:
...
+* An object of type [Tagged](https://github.com/apache/incubator-gobblin/blob/master/gobblin-metrics-libs/gobblin-metrics-base/src/main/java/org/apache/gobblin/metrics/Tagged.java) containing the tags for this metric context.
+* A `Set` of notification targets. Notification targets are objects of type [Function](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Function.html)<[Notification](https://github.com/apache/incubator-gobblin/blob/master/gobblin-metrics-libs/gobblin-metrics-base/src/main/java/org/apache/gobblin/metrics/notification/Notification.java), Void> which are all called every time there is a new notification. Notifications can be submitted to the Metric Context using the method `sendNotification(Notification)`. Notification targets can be added using `addNotificationTarget(Function<Notification, Void>)`.
```
Comment (csoseman) [inline gobblin-docs/metrics/Gobblin-Metrics-Architecture.md]: `Function` link should be https://google.github.io/guava/releases/15.0/api/docs/com/google/common/base/Function.html

File: gobblin-docs/user-guide/Gobblin-Schedulers.md [hunk 4]
```diff
@@ -44,7 +44,7 @@ This guide focuses on getting Gobblin to run in as a stand alone Java Process. T
...
+`gobblin-oozie-example-workflow.xml` contains an example Oozie workflow. This example simply launches a Java process that invokes the main method of the [`CliLocalJobLauncher`](https://github.com/apache/incubator-gobblin/blob/master/gobblin-runtime/src/main/java/org/apache/gobblin/runtime/local/CliLocalJobLauncher.java). The main method of this class expects two file paths to be passed to it (once again these files need to be on HDFS). The `jobconfig` arg should point to a file on HDFS containing all job configuration parameters. An example `jobconfig` file can be found [here](https://github.com/linkedin/gobblin/blob/master/gobblin-example/src/main/resources/wikipedia.pull). The `sysconfig` arg should point to a file on HDFS containing all system configuration parameters. An example `sysconfig` file for Oozie can be found [here](https://github.com/linkedin/gobblin/blob/master/gobblin-oozie/src/main/resources/local/gobblin-oozie-example-system.properties).
```
Comment (csoseman) [inline gobblin-docs/user-guide/Gobblin-Schedulers.md]: the Oozie config link should be changed to: https://github.com/apache/incubator-gobblin/blob/master/gobblin-oozie/src/test/resources/local/gobblin-oozie-example-system.properties

File: gobblin-docs/user-guide/Gobblin-on-Yarn.md [hunk 5]
```diff
@@ -71,37 +71,37 @@ This design makes it switch to a different resource manager, e.g., Mesos, by rep
...
 
 ## Gobblin WorkUnitRunner
 
 The WorkUnitRunner process runs the [`GobblinWorkUnitRunner`](https://github.com/linkedin/gobblin/blob/master/gobblin-yarn/src/main/java/gobblin/yarn/GobblinWorkUnitRunner.java), which uses a `ServiceManager` to manage the services supporting the operation of the WorkUnitRunner process. The services running in `GobblinWorkUnitRunner` will be discussed later. When it starts, the first thing `GobblinWorkUnitRunner` does is to connect to ZooKeeper and register itself as a Helix _participant_. It then starts the `ServiceManager`, which in turn starts the services it manages, as discussed below.
```
Comment (csoseman) [inline gobblin-docs/user-guide/Gobblin-on-Yarn.md]: It doesn't look like `GobblinWorkUnitRunner` exists in the current repo. We should probably ask the channel what happened to that and update the docs according.

Comment (jhsenjaliya) [reply gobblin-docs/user-guide/Gobblin-on-Yarn.md]: right, the only reference of this is in GobblinTaskRunner.java, which does what is described here, I will update to that.

**Comment by jhsenjaliya** [comment]
@csoseman Thanks for the review, i actually had to write little script to do these changes, since its not simple linkedIn to apache link replacement because the path of the file/module name has been changed.  anyway, expanded the scope to all invalid links, and has fixed it. Thanks

**Comment by jhsenjaliya** [comment]
In last commit, fixed all invalid file path in links and updated all possible LinkedIn links.

File: gobblin-docs/case-studies/Kafka-HDFS-Ingestion.md (hunk 1 above)
Comment (jhsenjaliya) [reply gobblin-docs/case-studies/Kafka-HDFS-Ingestion.md]: sure, i only targeted class files links, but i can expand this to all kinds of links

**Comment by jhsenjaliya** [comment]
Hi @abti , can you pls merge this, I want to change the docs on top of this as part of #2578 Thanks

**Comment by jhsenjaliya** [comment]
Thanks for taking this in.

**Pull request closed without merging**